### PR TITLE
Get rid of get_blobs_not_found and unify blob errors

### LINKED ISSUE #2628: Revisit `get_blobs_not_found` and blob related errors
Right now we might be having the blob errors down too many levels of abstraction. We probably need the caller to turn the error into the appropriate error. The idea of this task is to refactor the errors there so that we don't need a `get_blobs_not_found` function at all.
Comment with context https://github.com/linera-io/linera-protocol/pull/2610#discussion_r1800812358

### PR DESCRIPTION
**Pull request by ndr-ds**
## Motivation

We've had these two different error variants for blobs for a while: `BlobNotFoundOnRead` and `BlobsNotFound`. The motivation was to differentiate between errors when doing a storage read, and everything else. In the current state of the code, I don't think this distinction is really giving us much, and it complicates the code more.

We also have this `get_blobs_not_found` function that abstracts away the different nested error types.

## Proposal

Remove `BlobNotFoundOnRead`, and simplify things a bit. Wrote custom/manual `From` implementations for these different error types, so we can have a `BlobsNotFound` error everywhere and can get rid of `get_blobs_not_found`.
Fixes #2628

## Test Plan

CI

## Release Plan

- Nothing to do / These changes follow the usual release cycle.

### CODE REVIEW

**Comment by andresilva91** [comment]
* **#2649** <a href="https://app.graphite.dev/github/pr/linera-io/linera-protocol/2649?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @andresilva91 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: linera-execution/src/system.rs
Comment (ma2bd) [inline linera-execution/src/system.rs:401]: In the rest of the code, we try to normalize the errors during conversion instead

Comment (afck) [reply linera-execution/src/system.rs:401]: (There's also https://github.com/linera-io/linera-protocol/issues/2628 for this.)

File: linera-core/src/updater.rs
Comment (afck) [inline linera-core/src/updater.rs:288]: Is `BlobsNotFound` really guaranteed to contain all missing blobs?
If execution fails because some contract tries to read a missing blob, I assume this error contains only _that_ blob? It can't know about any other blobs that would have been encountered if execution had continued. So maybe we should keep this code here?

File: linera-execution/src/system.rs
Comment (andresilva91) [reply linera-execution/src/system.rs:401]: Yes, I agree this is not the ideal end solution for this, this is just an intermediate step until we get to #2628

File: linera-core/src/updater.rs
Comment (andresilva91) [reply linera-core/src/updater.rs:288]: Good point!

File: linera-execution/src/system.rs
Comment (afck) [reply linera-execution/src/system.rs:401]: Let's add a `TODO(#2628)` to both `get_blobs_not_found` functions.

Comment (andresilva91) [reply linera-execution/src/system.rs:401]: Talked with @afck a while back, I'm just gonna do #2628 here already

**Review by afck — APPROVED** [review]
I'm not sure about this. It mainly shows that we have too many error types that are nested in too many ways. But cleaning those up is beyond the scope of this PR, of course.

I'm also a bit concerned about the fact that we now convert `BlobsNotFound` to the corresponding variant of the higher-level type implicitly everywhere, using `?`: Should they really be treated all the same? E.g. aren't there cases where a validator should locally expect to have a particular blob—i.e. should respond with something like an "internal error"—that are different from other places where it's correct to respond with a `BlobsNotFound` to the client, so they can send them to us?

Anyway, I'm approving this for now because I don't have a simple better idea and I don't think it introduces _new_ problems, but I'd prefer if we could get another opinion on this PR, by a second reviewer.

**Comment by andresilva91** [comment]
Those are good points. I think right now though, AFAIU, the behavior is that we always return the error to the client, and fail on the client if the blob is actually not found. This is a behavior that could be looked into though, if we think it should be done differently 🤔

File: linera-core/src/client/mod.rs
Comment (afck) [inline linera-core/src/client/mod.rs:609]: Actually, I think this is a conversion we should _not_ make, since we will want to distinguish: If the error is that the _remote_ node doesn't have a blob, we need to send it. If the error is that _we_ don't have a blob, we need to request it or fail. (See https://github.com/linera-io/linera-protocol/pull/2871.)

File: linera-core/src/client/mod.rs
Comment (ma2bd) [inline linera-core/src/client/mod.rs:385]: this is weird but changed in the next PR, right?

Comment (ndr-ds) [reply linera-core/src/client/mod.rs:385]: I think @afck had a PR that fixed this behavior, so this is already gone in my local version of this PR. Will update it in a bit!

Comment (afck) [reply linera-core/src/client/mod.rs:385]: Didn't I already merge that? I think `try_download_blobs` returns `None` if it couldn't download all requested blobs? So I think you can remove this comparison.

Comment (ndr-ds) [reply linera-core/src/client/mod.rs:385]: I think that's what I said 😅 already have that done locally, but haven't updated the PR yet

**Review by afck — APPROVED** [review]

**Comment by ndr-ds** [comment]
### Merge activity

* **Nov 13, 3:58 PM EST**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/linera-io/linera-protocol/2649).

**Pull request merged**
